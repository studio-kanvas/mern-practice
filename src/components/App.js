import React, {PropTypes} from 'react';
import {connect} from 'react-redux';
import {BrowserRouter as Router, Route, Switch} from 'react-router-dom';
import HeaderNav from './home/HeaderNav';
import PageNotFound from './common/PageNotFound';
import Home from './home/Home';
import CourseListContainer from './course/CourseListContainer'; // eslint-disable-line import/no-named-as-default
import AddOrEditCourseContainer from './course/AddOrEditCourseContainer'; // eslint-disable-line import/no-named-as-default
import About from './About';
import createBrowserHistory from 'history/createBrowserHistory';



const history = createBrowserHistory();



class App extends React.Component {
    render() {
        return (
            <div >
                <Router history={history}>
                    <div>
                        <HeaderNav apiCallsInProgress={this.props.apiCallsInProgress}/>
                        <Switch>
                            <Route exact path="/" component={Home}/>
                            <Route path="/courses" component={CourseListContainer}/>
                            <Route exact path="/course" component={AddOrEditCourseContainer}/>                
                            <Route path="/course/:id" component={AddOrEditCourseContainer}/>        
                            <Route path="/about" component={About}/>
                            <Route component={PageNotFound}/>
                        </Switch>                        
                    </div>                        
                </Router>
            </div>
        );
  }
}



App.propTypes = {
  apiCallsInProgress: PropTypes.bool.isRequired
};


function mapStateToProps(state) {
    return {
        apiCallsInProgress: state.apiReducer.apiCallsInProgress > 0
    };
}



export default connect(mapStateToProps)(App);

