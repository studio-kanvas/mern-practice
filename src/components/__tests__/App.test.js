import React from 'react';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import createBrowserHistory from 'history/createBrowserHistory';
import App from '../App';


describe('App.test.js', () => {
    let history;

    beforeEach(() => {
      history = createBrowserHistory();
    });

    it('renders without crashing', () => {
        const props = { history: history };

        const wrapper = shallow(<App {...props}/>);

        expect(wrapper.length).toEqual(1);

        const tree = toJson(wrapper);
        expect(tree).toMatchSnapshot();
    });
});
