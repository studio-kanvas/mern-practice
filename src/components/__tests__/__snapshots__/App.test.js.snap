exports[`App.test.js renders without crashing 1`] = `
<div>
  <BrowserRouter
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "key": undefined,
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }>
    <div>
      <Connect(HeaderNavContainer) />
      <Switch>
        <Route
          component={[Function]}
          exact={true}
          path="/" />
        <Route
          component={[Function]}
          path="/courses" />
        <Route
          component={[Function]}
          exact={true}
          path="/course" />
        <Route
          component={[Function]}
          path="/course/:id" />
        <Route
          component={[Function]}
          path="/about" />
        <Route
          component={[Function]} />
      </Switch>
    </div>
  </BrowserRouter>
</div>
`;
